packages/console/src/containers/ConsoleContent/hooks.ts

```diff
@@ -0,0 +1,68 @@
+import { Prompt, useLogto } from '@logto/react';
+import { getTenantOrganizationId } from '@logto/schemas';
+import { useContext, useEffect, useState } from 'react';
+
+import { TenantsContext } from '@/contexts/TenantsProvider';
+import useCurrentTenantScopes from '@/hooks/use-current-tenant-scopes';
+import useRedirectUri from '@/hooks/use-redirect-uri';
+import { saveRedirect } from '@/utils/storage';
+
+const useTenantScopeListener = () => {
+  const { currentTenantId, isInitComplete } = useContext(TenantsContext);
+  const { isAuthenticated, clearAccessToken, clearAllTokens, getOrganizationTokenClaims, signIn } =
+    useLogto();
+  const [tokenClaims, setTokenClaims] = useState<string[]>([]);
+  const [isDecodingClaims, setIsDecodingClaims] = useState(false);
+  const redirectUri = useRedirectUri();
+  const { scopes = [], isLoading } = useCurrentTenantScopes();
+
+  useEffect(() => {
+    (async () => {
+      if (!isAuthenticated || !isInitComplete) {
+        return;
```

Comment:
the main purpose of putting the hook into this directory is to avoid these checks?